### PR TITLE
[fix] Virtualize the run params list in the Run page

### PR DESCRIPTION
**Pull request by roubkar**
- Implemented custom `DictVisualizer` which has built-in support for virtualization to replace `react-json-view`
- Added a helper function to visualize data type
- Added a helper function to color values based on their type

### CODE REVIEW

**Comment by roubkar** [comment]
Resolves https://github.com/aimhubio/aim/issues/1994

File: aim/web/ui/src/components/kit/DictVisualizer/DictVisualizer.d.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import { CSSProperties } from 'react';
+
+export interface IDictVisualizerProps {
+  src: object | { [key: string]: unknown };
```
Comment (VkoHov) [inline aim/web/ui/src/components/kit/DictVisualizer/DictVisualizer.d.ts:4]: maybe we can use Record<string, unknown>

File: aim/web/ui/src/components/kit/DictVisualizer/DictVisualizer.scss [hunk 2]
```diff
@@ -0,0 +1,57 @@
+@use 'src/styles/abstracts' as *;
+
+.DictVisualizer {
+  font-family: "Inconsolata", monospace;
```
Comment (VkoHov) [inline aim/web/ui/src/components/kit/DictVisualizer/DictVisualizer.scss:4]: let's use the monospaceFontFamily mixin

File: aim/web/ui/src/components/kit/DictVisualizer/DictVisualizer.tsx [hunk 3]
```diff
@@ -0,0 +1,309 @@
...
+  IDictVisualizerRowProps,
+} from './DictVisualizer.d';
+
+import './DictVisualizer.scss';
+
+const ROW_SIZE = 22;
+
+function DictVisualizer(props: IDictVisualizerProps) {
+  const [collapsedItems, setCollapsedItems] = React.useState<{
+    [key: string]: boolean;
```
Comment (VkoHov) [inline aim/web/ui/src/components/kit/DictVisualizer/DictVisualizer.tsx:27]: Record<string, boolean>

File: aim/web/ui/src/components/kit/DictVisualizer/DictVisualizer.tsx [hunk 4]
```diff
@@ -0,0 +1,309 @@
...
+      level: number = 0,
+      parentKey: string = 'root',
+    ) => {
+      let rows: DictVisualizerRowType[] = [];
+
+      // Add top level brackets
+      if (level === 0) {
+        if (Array.isArray(dict)) {
+          let nestedItemsLength = dict.length;
+          rows.push({
```
[the remark this answers is not in the document]
Comment (roubkar) [reply aim/web/ui/src/components/kit/DictVisualizer/DictVisualizer.tsx:43]: yeah, but the function will be of the same size, right? you still will need to pass all this object properties

**Review by arsengit — APPROVED** [review]

**Review by VkoHov — APPROVED** [review]
lgtm!

**Pull request merged**